pkg/sbproxy/notifications/producer_test.go

```diff
@@ -219,6 +269,69 @@ var _ = Describe("Notifications", func() {
 			})
 		})
 
+		Context("When a server that accept mTLS exists", func() {
+			var tlsServer *httptest.Server
+			var tlsSocketServer *wsServer
+			BeforeEach(func() {
+				tlsServer, tlsSocketServer = newServerWithTLS()
+				producerSettings = &notifications.ProducerSettings{
+					MinPingPeriod:        100 * time.Millisecond,
+					ReconnectDelay:       100 * time.Millisecond,
+					PongTimeout:          20 * time.Millisecond,
+					ResyncPeriod:         300 * time.Millisecond,
+					PingPeriodPercentage: 60,
+					MessagesQueueSize:    10,
+				}
+			})
+
+			AfterEach(func() {
+				tlsServer.Close()
+			})
+
+			When("valid mTLS certificate settings are used", func() {
+
+				BeforeEach(func() {
+					smSettings = &sm.Settings{
+						URL:                  tlsServer.URL,
+						User:                 "admin",
+						Password:             "admin",
+						RequestTimeout:       11 * time.Second,
+						TLSClientCertificate: ClientCertificate,
+						TLSClientKey:         ClientKey,
+						SkipSSLValidation:    true,
+						NotificationsAPIPath: "/v1/notifications",
+					}
+				})
+
+				It("successfully pings the mTLS server using valid client certificate", func(done Done) {
+					times := make(chan time.Time, 10)
+					tlsSocketServer.onClientConnected = func(conn *websocket.Conn) {
+						times <- time.Now()
+					}
+					tlsSocketServer.pingHandler = func(string) error {
+						defer GinkgoRecover()
+						now := time.Now()
+						times <- now
+						if len(times) == 3 {
+							start := <-times
+							for i := 0; i < 2; i++ {
+								t := <-times
+								pingPeriod, err := time.ParseDuration(tlsSocketServer.maxPingPeriod)
+								Expect(err).ToNot(HaveOccurred())
+								Expect(t.Sub(start)).To(BeNumerically("<", pingPeriod))
+								start = t
+							}
+							close(done)
+						}
+						tlsSocketServer.conn.WriteControl(websocket.PongMessage, []byte{}, now.Add(1*time.Second))
+						return nil
+					}
+					newProducer, _ := notifications.NewProducer(producerSettings, smSettings)
+					newProducer.Start(producerCtx, group)
+				}, (500 * time.Second).Seconds())
```

Comment:
500 seconds? do we really need to wait that long?

Reply:
I played with it to debug, we don't need to wait here at all .. removing :)